### DISCUSSION TITLE
Is there a way to make renaming a file start with no text other then the extension?

### DISCUSSION THREAD

**Discussion by SafriXV** [sxyazi/yazi#1079] [1. Q&A] [answered]
In most gui file managers (I specifically remember thunar having this feature), pressing f2 on a file highlights the name of a file (excluding the extension) to be replaced by whatever you type next. I frequently have to rename subtitle files to match the video file they correspond to for mpv to pick up on. Being able to mimic this behavior in yazi would be really nice. 

I imagine assigning a keybind to execute a shell command that starts with "rename" would work but I'm not sure to do this. Built in shell keybinding wouldn't fit this use case because the file names rarely ever don't have spaces in them.

Any help would be appreciated!

**Comment by sxyazi** [comment, marked as answer]
You can bind a key for example `R` to do this:

```
{ on = [ "R" ], run = "rename --empty=stem --cursor=start" }
```

When you click `R`, it will empty the filename then move the cursor to start. See https://yazi-rs.github.io/docs/configuration/keymap#manager.rename for details.

**Reply by SafriXV** [reply]
Works perfectly, thank you.
